src/main/scala/Top.scala

```diff
@@ -0,0 +1,44 @@
+package flexpret
+import chisel3._
+import flexpret.core.{Core, FlexpretConfiguration, GPIO, HostIO}
+import flexpret.Wishbone.{HelloWishbone, WishboneMaster}
+
+case class TopConfig(
+  coreCfg : FlexpretConfiguration
+)
+
+
+class TopIO(topCfg: TopConfig) extends Bundle {
+  val gpio = new GPIO()(topCfg.coreCfg)
+}
+
+class Top(topCfg: TopConfig) extends MultiIOModule {
+  val io = IO(new TopIO(topCfg))
+  // FlexPret core
+  val core = Module(new Core(topCfg.coreCfg))
+
+  // WB Master connecting FP to memory mapped devices
+  val wbMaster = Module(new WishboneMaster(topCfg.coreCfg.busAddrBits)(topCfg.coreCfg))
+  core.io.bus <> wbMaster.busIO
+
+  // Simple Wishbone device for testing
+  val wbDevice = Module(new HelloWishbone())
```

Comment:
Is this entire class just for testing?

Reply:
This is the top level module for a SoC with one or several FlexPRET CPUs